### ISSUE TITLE
Fatal Error - Fails to parse csv when duplicate record in imported csv .

### ISSUE THREAD

**Issue by jcop007** [silverstripe/silverstripe-redirectedurls#35]
Hi
We're importing a csv with cols 'FromBase','FromQuerystring','To'
If any of the rows are not unique across the two cols 'FromBase' & 'FromQuerystring' it will throw a fatal error.  
Failed to parse C:\WINDOWS\TEMP\silverstripe-cache-php5.5.12C--wamp-www-enjoyillinoiscom\JONATHAN-4/BulkLoader589cbd0d829a31.37719436.csv 
Fatal error: Call to a member function CreatedCount() on a non-object in C:\wamp\www\enjoyillinoiscom\framework\admin\code\ModelAdmin.php on line 445
Thanks

**Comment by dhensby** [comment]
Is this something you'd be happy to investigate?

**Comment by michalkleiner** [comment]
@jcop007 there is a unique database index on those two columns. It makes sense to limit it so there's only one pair of FromBase and FromQueryString. If there was more combinations, how would you decide which destination to choose to redirect to?

**Comment by jcop007** [comment]
@michalkleiner Yes, that's true.  My main issue was that it took use many hours to figure out what was causing the error and then which row in the spreadsheet was responsible.  If the error message could display that information it would be very helpful.

**Comment by dhensby** [comment]
hmm, that is an unhelpful error if the real problem is a duplicate unique index - anyone prepared to patch it?

**Comment by stevie-mayhew** [comment]
I've just tested this against the 4.x compatible line (master) and it is no longer an issue.

@jcop007 do you have a fix for the 3.x compatible (1) line?

**Comment by jcop007** [comment]
@stevie-mayhew No, we didn't write a fix.  We just removed duplicates once we figured out that was the issue.  I'll take another look and see if we can get a fix for it.

**Comment by priyashantha** [comment]
@stevie-mayhew @dhensby Here is a PR https://github.com/silverstripe/silverstripe-redirectedurls/pull/59 for this issue. Could someone please review?
